### ISSUE TITLE
`Select` initial value is out of view when first expanded

### ISSUE THREAD

**Issue by TomJGooding** [Textualize/textual#5255]
When the `Select` widget is first expanded, the initial selected value is out of view when it appears further down in the list of options.

```python
from textual.app import App, ComposeResult
from textual.widgets import Select


class SelectApp(App):
    def compose(self) -> ComposeResult:
        yield Select.from_values(
            range(20),
            value=15,
        )


if __name__ == "__main__":
    app = SelectApp()
    app.run()
```

**Comment by TomJGooding** [comment]
I think the problem is when the `Select` is first expanded, where this calls `OptionList.scroll_to_highlight` the region values here are all zeros:

https://github.com/Textualize/textual/blob/8d99130708b3a862fc849884e0015c2c864345d1/src/textual/widgets/_option_list.py#L891-L896

**Comment by TomJGooding** [comment]
After testing with the latest version (0.88.1), expanding the `Select` with the keyboard now seems to works as expected, but not when clicked with the mouse.

Maybe rather than `self.call_after_refresh` added on L520, should it be `overlay.call_after_refresh`...?

**Comment by willmcgugan** [comment]
That does appear to be a fix!

**Comment by willmcgugan** [comment]
But I can replicate the issue in the demo. Curious.

**Comment by TomJGooding** [comment]
It looks like this was resolved in #5314. 

I noticed a difference in the scroll behaviour when first expanded with the mouse vs the keyboard, but that probably should be a new issue.
